internal/datastore/access_link.go

```diff
@@ -0,0 +1,90 @@
+package datastore
+
+import (
+	"context"
+	"database/sql"
+
+	"github.com/delphis-inc/delphisbe/graph/model"
+	"github.com/sirupsen/logrus"
+)
+
+func (d *delphisDB) GetAccessLinkBySlug(ctx context.Context, slug string) (*model.DiscussionAccessLink, error) {
+	logrus.Debug("GetAccessLinkBySlug::SQL Query")
+	if err := d.initializeStatements(ctx); err != nil {
+		logrus.WithError(err).Error("GetAccessLinkBySlug::failed to initialize statements")
+		return nil, err
+	}
+
+	dal := model.DiscussionAccessLink{}
+	if err := d.prepStmts.getAccessLinkBySlugStmt.QueryRowContext(
+		ctx,
+		slug,
+	).Scan(
+		&dal.DiscussionID,
+		&dal.LinkSlug,
+		&dal.CreatedAt,
+		&dal.UpdatedAt,
+		&dal.DeletedAt,
+	); err != nil {
+		if err == sql.ErrNoRows {
+			return &model.DiscussionAccessLink{}, nil
+		}
+		logrus.WithError(err).Error("failed to execute getAccessLinkBySlugStmt")
+		return nil, err
+	}
+
+	return &dal, nil
+}
+
+func (d *delphisDB) GetAccessLinkByDiscussionID(ctx context.Context, discussionID string) (*model.DiscussionAccessLink, error) {
```

Comment:
Could there be multiple?

Reply:
there should not be more than one active at a time. I am adding a `limit 1` just in case.